misc/sample_application/uservm/Makefile

```diff
@@ -6,7 +6,7 @@ CFLAGS = -Wall -Wextra -Wabi=11 -pedantic
 LDLIBS = -lrt -pthread
 
 all: userApp.cpp
-	$(CP) $(CFLAGS) -o userapp userApp.cpp ivshmemlib.c $(LDLIBS)
+	$(CP) $(CFLAGS) -o ../build/userapp userApp.cpp ivshmemlib.c $(LDLIBS)
```

Comment:
Ditto

Reply:
@gvancuts 
1 . Modify the Makefile in the parent directory pass the OUT_DIR to the sub-makefies.
2. Modify the Makefile in the sub-directory in case the user just run the make command in the sub-directory.
Fore update the commit in this PR